File: src/leetcode/100-149/128.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        seen : set[int] = set(nums)
        acc : int = 0

        def streak(number: int) -> int:
            acc = 1

            while number + 1 in seen:
                number, acc = number + 1, acc + 1

            return acc

        for num in seen:
            # Beginning of the sequence
            if num - 1 not in seen:
                acc = max(acc, streak(number=num))

        return acc

File: cpp/128.cpp
#include <unordered_set>

inline int streak(const std::unordered_set<int>& set, int number) {
  int acc = 1;

  while (set.contains(number + 1)) {
    acc += 1;
    number += 1;
  }

  return acc;
}

class Solution {
public:
  int longestConsecutive(std::vector<int> &nums) {

    std::unordered_set<int> set = std::unordered_set<int>(nums.begin(), nums.end());
    int acc = 0;

    for (const int num : set) {
      if (!set.contains(num - 1)) {
        acc = std::max(acc, streak(set, num));
      }
    }

    return acc;
  }
};